File: python/数组/228. 汇总区间.py
from typing import List
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        i = 0
        res = []
        while i < len(nums):
            start = nums[i]
            while i < len(nums)-1 and nums[i+1] - nums[i] == 1:
                i+=1
            end = nums[i]
            if start == end:
                res.append(str(start))
            else:
                res.append("%d->%d" % (start,end))
            i+=1
        return res
s = Solution()
nums = [0,1,2,4,5,7]
print(s.summaryRanges(nums))

File: go/数组/228. 汇总区间.go
package array

import (
	"fmt"
	"strconv"
)

func summaryRanges(nums []int) []string {
	var res []string
	for i := 0; i < len(nums); i++ {
		start := nums[i]
		for i < len(nums)-1 && nums[i+1]-nums[i] == 1 {
			i++
		}
		end := nums[i]
		if start == end {
			res = append(res, strconv.Itoa(start))
		} else {
			res = append(res, fmt.Sprintf("%d->%d", start, end))
		}
	}
	return res
}
